sharpy/postproc/liftdistribution.py

```diff
@@ -1,79 +1,131 @@
         settings.to_custom_types(self.settings, self.settings_types, self.settings_default)
-        self.ts_max = len(self.data.structure.timestep_info)
         self.caller = caller
+        self.folder = data.output_folder
+        if not os.path.exists(self.folder):
+            os.makedirs(self.folder)
 
     def run(self, online=False):
-        if not online:
-            for self.ts in range(self.ts_max):
-                self.lift_distribution()
-            cout.cout_wrap('...Finished', 1)
-        else:
-            self.ts = len(self.data.structure.timestep_info) - 1
-            self.lift_distribution()
+        self.lift_distribution(self.data.structure.timestep_info[self.data.ts],
+                               self.data.aero.timestep_info[self.data.ts])
         return self.data
 
-    def lift_distribution(self):
-        # add entry to dictionary for postproc
-        tstep = self.data.structure.timestep_info[self.data.ts]
-        tstep.postproc_cell['lift_distribution'] = init_matrix_structure(dimensions=tstep.dimensions,
-                                                                           with_dim_dimension=False)
-
-        norm = 1.0
-
-        for i_surf in range(tstep.n_surf):
-            n_dim, n_m, n_n = tstep.zeta[i_surf].shape
-
-            forces = tstep.forces[i_surf] + tstep.unsteady_forces[i_surf]
-            abs_forces = np.zeros((n_m, n_n))
-            chord = np.zeros((n_n,))
-            for i_n in range(n_n):
-                chord[i_n] = np.abs(tstep[i_surf].zeta[:, -1, i_n] - tstep[i_surf].zeta[:, 0, i_n])
-                if not i_surf and not i_n:
-                    if self.settings['normalise']:
-                        norm = chord[0]
-
-                for i_m in range(n_m):
-                    abs_forces[i_n, i_m] = np.abs(forces[:, i_m, i_n])
-
-                tstep.postproc_cell['lift_distribution'][i_surf][i_n, i_m] = np.sum(abs_forces[i_n, :], axis = 2)/norm
+    def lift_distribution(self, struct_tstep, aero_tstep):
+        # Force mapping
+        rot = algebra.quat2rotation(struct_tstep.quat)
+        forces = mapping.aero2struct_force_mapping(
+            aero_tstep.forces + aero_tstep.dynamic_forces,
+            self.data.aero.struct2aero_mapping,
+            aero_tstep.zeta,
+            struct_tstep.pos,
+            struct_tstep.psi,
+            self.data.structure.node_master_elem,
+            self.data.structure.connectivities,
+            struct_tstep.cag(),
+            self.data.aero.aero_dict)
+        # Prepare output matrix and file 
+        N_nodes = self.data.structure.num_node
+        numb_col = 4
+        header= "x,y,z,fz"
+        # get aero forces
+        lift_distribution = np.zeros((N_nodes, numb_col))
+        for inode in range(N_nodes):
+            if self.data.aero.aero_dict['aero_node'][inode]:
+                # transform forces from B to A frame
+                lift_distribution[inode,3]=np.dot(rot.T, forces[inode, :3])[2]  # lift force
```

Comment:
Hmm I believe this will not always be the lift force. In the case, say the fluid is rotated but the wing is straight (i.e. A and G coincident) you will not be capturing the perpendicular to the free stream.

How @ArturoMS13 implemented this was to find the free stream direction at the section and project the force onto it (this would be the drag), then, the remainder of the force is assumed to be the lift (there would be a side force component particularly near tips but it makes the calculation simpler). 

If you have a look at the Polars file you may it is done there if you'd like to implement it

Reply:
Also, if I'm not mistaken `forces` is in B frame. To get them to `A` you'll need the local CRV to find `cab` and then you can use `rot.T` to get them onto G.

This is always so confusing......it's been years for me and still have doubts. I put a little function in `algebra` called `get_transformation()` that gets you the projection matrix based on the string you give it.